#!/bin/sh -e
## 20-honour-papersize.dpatch, by Finn-Arne Johansen <faj@bzz.no>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patched to support /etc/papersize   

if [ $# -lt 1 ]; then
echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
      -patch) patch $patch_opts -p1 < $0;;
      -unpatch) patch $patch_opts -p1 -R < $0;;
      *)
      echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
      exit 1;;
esac

exit 0

@DPATCH@
--- foo2zjs.clean/foo2hp2600-wrapper.in	2005-11-26 23:04:38.000000000 +0100
+++ foo2zjs-20070624/foo2hp2600-wrapper.in	2005-11-26 23:16:26.000000000 +0100
@@ -155,7 +155,8 @@
 
 MEDIA=1
 COPIES=1
-PAPER=1
+test -r /etc/papersize && PAPER=$(cat /etc/papersize)
+test "$PAPER" || PAPER=1
 RES=600x600
 SOURCE=7
 NUP=
diff -ru ../foo2zjs.clean/foo2oak-wrapper.in ./foo2oak-wrapper.in
--- ../foo2zjs.clean/foo2oak-wrapper.in	2005-11-26 23:04:38.000000000 +0100
+++ ./foo2oak-wrapper.in	2005-11-26 23:15:13.000000000 +0100
@@ -122,7 +122,8 @@
 QUALITY=1
 MEDIA=1
 COPIES=1
-PAPER=1
+test -r /etc/papersize && PAPER=$(cat /etc/papersize)
+test "$PAPER" || PAPER=1
 RES=600x600
 SOURCE=7
 NUP=
diff -ru ../foo2zjs.clean/foo2zjs-wrapper.in ./foo2zjs-wrapper.in
--- ../foo2zjs.clean/foo2zjs-wrapper.in	2005-11-26 23:04:38.000000000 +0100
+++ ./foo2zjs-wrapper.in	2005-11-26 23:16:28.000000000 +0100
@@ -160,7 +160,8 @@
 QUALITY=1
 MEDIA=1
 COPIES=1
-PAPER=1
+test -r /etc/papersize && PAPER=$(cat /etc/papersize)
+test "$PAPER" || PAPER=1
 RES=1200x600
 SOURCE=7
 NUP=
